src/mappers/AppUserMapper.ts

```diff
@@ -0,0 +1,65 @@
+import { EventSignInRequest, EventSignInResponse } from '@Payloads';
+import { AppUser } from '@Entities';
+import { AppUserRepositoryToken } from '@Repositories';
+
+import { Repository } from 'typeorm';
+import { classToPlain, plainToClass } from 'class-transformer';
+import { singleton, inject } from 'tsyringe';
+
+@singleton()
+export class AppUserMapper {
+  private appUserRepository: Repository<AppUser>;
+
+  constructor(@inject(AppUserRepositoryToken) appUserRepository: Repository<AppUser>) {
+    this.appUserRepository = appUserRepository;
+  }
+
+  /**
+   * Converts an EventSignInRequest payload to an AppUser entity and
+   * returns the newly created entity to the caller.
+   *
+   * @param appUserRequest The request payload from which the AppUser entity
+   *                       is created.
+   */
+  requestToNewEntity(appUserRequest: EventSignInRequest): AppUser {
+    const plainAppUserRequest: Object = classToPlain(appUserRequest);
+    return this.appUserRepository.create(plainAppUserRequest);
+  }
+
+  /**
+   * Updates an existing AppUser entity with new data from the request payload.
+   * If there is no AppUser entity with the passed-in appUserId, then return undefined.
+   *
+   * @param appUserRequest The request payload from which the updated existing AppUser
+   *                       entity is created
+   * @param appUserId The supposed ID of an existing AppUser entity.
+   */
+  async requestToExistingEntity(
+    appUserRequest: EventSignInRequest,
+    appUserId: number
+  ): Promise<AppUser | undefined> {
+    const appUserObj: AppUser = appUserRequest as AppUser;
+    appUserObj.id = appUserId;
+
+    const appUser: AppUser = await this.appUserRepository.preload(appUserObj);
+    if (appUser == undefined) {
```

Comment:
can probably just return appUser

Reply:
You mean no null check?

Reply:
oh that's == undefined not === undefined lol

preload - Creates a new entity from the given plain javascript object. If the entity already exists in the database, then it loads it (and everything related to it), replaces all values with the new ones from the given object, and returns the new entity. The new entity is actually an entity loaded from the database with all properties replaced from the new object.
Note that given entity-like object must have an entity id / primary key to find entity by. Returns undefined if entity with given id was not found.

Reply:
can probably just return it since preload returns undefined